common/src/stack/command/stack/commands/argument_processors/firmware.py

```diff
@@ -0,0 +1,330 @@
+from pathlib import Path
+from collections import namedtuple
+from stack.exception import CommandError
+
+class FirmwareArgumentProcessor:
+	"""A mixin to process firmware command arguments."""
+
+	def remove_duplicates(self, args):
```

Comment:
I think `remove_duplicates` is more descriptive than `tuple(set(thing))` so I'll probably keep that around unless my arm is twisted into submission.

I will look into this order preserving requirement though since I didn't know that was a thing. Might end up having to stay as a function in that case anyways depending on the logic required to preserve order.

Reply:
Changes made to preserve ordering @bsanders.